### PR TITLE
new feature: crop to selection

### PR DESCRIPTION
**Pull request by beetrootpaul**
In this PR I implemented a "Crop to selection" feature. Example use case: canvas contain multiple version of some logo, and you want to marquee one of them and export as a PNG upscaled a lot.

I organised my changes into kinda isolated commits. Take a look at them and their messages in order to have better understanding of what has happened in this PR.

### CODE REVIEW

File: extension/super-export.lua [hunk 1]
```diff
@@ -27,6 +27,32 @@ local function create_error(str, dialog, exit)
     if (exit == 1) then dialog:close() end
 end
 
+local function calculate_export_size(original_width, original_height, resize_percentage)
+    return {
+        width = original_width * resize_percentage / 100,
```
Comment (david-fletcher) [inline extension/super-export.lua:32]: would you mind adding the parentheses back in here to indicate order of operations?
image

**Review by david-fletcher — CHANGES_REQUESTED** [review]
This looks great! If you don't mind adding some parentheses in that one function, I'll merge this PR in and then get the release prepared! Thanks again for doing this!

File: extension/super-export.lua (hunk 1 above)
Comment (beetrootpaul) [reply extension/super-export.lua:32]: no problem 👍 BTW Why does this matter? I mean, I even learnt in the past (but in another programming language with different data type for numbers, I suppose) to multiple first, then divide, not the other way 🤔 

(commit in preparation)

Comment (beetrootpaul) [reply extension/super-export.lua]: added in https://github.com/david-fletcher/super-export/pull/2/commits/0b3a3b7e02b0aa61486c0b6894ac28a21b052a63

**Review by david-fletcher — APPROVED** [review]
great work!

**Pull request merged**